### PR TITLE
Encoding/tl

### PR DESCRIPTION
**Pull request by zweihander**

### CODE REVIEW

File: encoding/tl/encoder.go [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+
+				if err := encodeValue(newCursor, fieldVal); err != nil {
+					return fmt.Errorf("field '%s': %w", vtyp.Field(i).Name, err)
+				}
+			}
+
+			continue
+		}
+
+		// FIXME:
```
Comment (ololosha228) [inline encoding/tl/encoder.go:119]: нулевое значение у филда может быть абсолютно спокойно, тут скорее проблема именно со структурами, если она нулевая, то это повод вкинуть ошибку

false тоже IsZero, но при этом он может кодироваться как надо

Comment (zweihander) [reply encoding/tl/encoder.go]: исправлено в https://github.com/xelaj/mtproto/pull/40/commits/84d0c8c646cff4604daa2a2de30c765d530410eb

**Comment by ololosha228** [comment]
@zweihander  TL encoder package merged into master. Currently no on of the current packages are not affected to this change, so it can be safely add into master.

**Comment by ololosha228** [comment]
@zweihander encoding/tl is fully implemented and merged into master. Next step is rework rest of mtproto code.

**Comment by ololosha228** [comment]
We are sooooooo close to 1.0!

I made [this](https://github.com/xelaj/mtproto/tree/new_decoder_refactoring) branch, which will be merged into master soon!

**Comment by ololosha228** [comment]
@zweihander i'll close this PR, cause we have already #45 

Most of this PR is merged there. Leave a feedback please!

Also, i mentioned you in comments btw.

**Pull request closed without merging**